src/ui/duke_script_runner.cpp

```diff
@@ -233,33 +235,74 @@ void DukeScriptRunner::handleKeyboardEvent(const SDL_Event& event) {
 }
 
 void DukeScriptRunner::handleGameControllerEvent(const SDL_Event& event) {
-  if (event.type != SDL_CONTROLLERBUTTONDOWN) {
+  if (!hasMenuPages()) {
     return;
   }
 
-  // Menu interaction using dpad, button B and A
```

Comment:
This file can get refactored now. All non important functionality can be moved to a different file, that will shorten its size and reduce complexity due to obscurity issue

Reply:
It's a pretty big file, that's true. I have to admit that I don't see any immediately obvious way to split it up though. All of this code is required to implement the features of DukeScript. There's not much in terms of "non important" functionality here.

I can also see how this might be an odd/unexpected place to find the code for menu navigation, but that's because the original game uses DukeScript to implement its menus.

What I _could_ imagine is introducing a helper class to deal with the input handling. Something like a `MenuInputHandler`. And then other places in the code like `DukeScriptRunner` or elsewhere would talk to that class instead of handling SDL events directly. That would make it easier to change input handling without having to change a lot of different places in the code.